### ISSUE TITLE
Allow triagebot to create hackmds

### ISSUE THREAD

**Issue by jackh726** [rust-lang/triagebot#1740]
It would be great to have triagebot auto-generate meeting hackmds. HackMD has a (newish?) API: https://hackmd.io/@hackmd-api/developer-portal/https%3A%2F%2Fhackmd.io%2F%40hackmd-api%2FrkoVeBXkq

This might needs some @rust-lang/infra work too, since there aren't "organization" API tokens. So, we would likely have to have a triagebot user, then add that user to any teams that we want to do any automation with.

**Comment by Mark-Simulacrum** [comment]
I'm happy to provision a token and wire it up into an env variable for triagebot to read at runtime.

**Comment by apiraino** [comment]
I've been using HackMD through API extensively for ~2 years with a simple Python script, namely using the [team notes API](https://hackmd.io/@hackmd-api/team-notes-api). I thought about integrating HackMD into the triagebot meetings agenda generator(s) but I was always a bit hesitant because there are some issues with this service.

Some examples:
- The [`/update`](https://hackmd.io/@hackmd-api/team-notes-api#Update-a-note-in-a-Team%E2%80%99s-workspace) endpoint is weird. It sends the complete Markdown file but rarely the page of the HackMD document shows the full document uploaded. I need to repeatedly upload the same document over and over (!) to see it incrementally complete on the website.
- The same `/update` API could also be used to change read/write permissions to a document. This part does not work.
- Other bugs in the past appeared and disappeared. AFAIK there is no changelog or releases, so hard to follow the development.
- Tried once contacting their support (an email contact), never received a reply.

Just my .2 cents, hope I don't sound too pessimistic :slightly_smiling_face: . All I am saying is that an HTTP client should work around these bugs.

**Comment by jackh726** [comment]
Yeah, I imagined it would take a little bit of tinkering to get this "right".

@apiraino maybe we should chat about your experience here sometime and discuss your thoughts. If this is something that we want (as a Project) and there currently problems with the API, then maybe a more "official" email could help.

Idk, we should talk!
